C_CPP/Data Structures/User Defined Data Structures/Fenwick Tree/Fenwick_Tree.md

```diff
@@ -0,0 +1,52 @@
+<h1 align="center">  Fenwick Tree or Binary Indexed Tree </h1>
+
+A Fenwick Tree is a user-defined data structure that helps to calculate Prefix sums or cumulative frequency tables in a very efficient manner.It is a representational form of an array and can query or update elements in O(logN) time complexity.It is also known as a Binary Indexed Tree as it makes use of bit manipulation concepts for implementation purposes.
+
+Need:
+
+- We can perform both Query and Update operations even in a large array in O(logN) time using Fenwick Trees.
+- Fenwick Trees require less space as compared to Segment Tress.
+- They have an easy implementation code which is useful during various programming contests.
+
+Terminology:
+
+- Query:Operation performed on an interval or on an element.For example minimum or maximum or summation in a particular range.
+- Update:To update the fenwick tree.There are two types of updates in a fenwick tree.
+  - Point Updates:Updating a particular element of the binary indexed tree with a value.
+  - Range Updates:Executing queries on a range of elements with a value.
+
+## Bit Manipulation Concepts:
+
+The basic concept behind Fenwick Tree is that we can represent all positive integers as the sum of powers of 2.Example: 21 can be represented as 21 = 16 + 4 + 1 = 2^4 +2^2 +2^0.This is the reason why Fenwick Tree is also referred to as the Binary Indexed Tree.
```

Comment:
```suggestion
The basic concept behind Fenwick Tree is that we can represent all positive integers as the sum of powers of 2.  
Example: 21 can be represented as 21 = 16 + 4 + 1 = 2<sup>4</sup> +2<sup>2</sup> +2<sup>0</sup>.  
This is the reason why Fenwick Tree is also referred to as the Binary Indexed Tree.
```

Reply:
@shreya024  This one is optional though...
If you don't feel like using sup tags then just embolden the numerical text